staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -106,7 +106,21 @@ func lookupServiceName(name string) (EgressType, error) {
 }
 
 func tunnelHTTPConnect(proxyConn net.Conn, proxyAddress, addr string) (net.Conn, error) {
-	fmt.Fprintf(proxyConn, "CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", addr, "127.0.0.1")
+	addrUrl, err := url.Parse(addr)
+	if err != nil {
+		return nil, err
+	}
+	connectReq := &http.Request{
```

Comment:
nit: Would you mind including a comment that this is over HTTP 1.1?

Reply:
Oh, do we need to force to use of http 1.1 instead of adding a comment? If necessary, I can modify the Host on the basis of the original lline.

```
fmt.Fprintf(proxyConn, "CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", addr, addrUrl.Host)
```

Reply:
https://golang.org/pkg/net/http/#Request.Write

The http library write defaults to HTTP 1.1 so that's probably unnecessary. 

/cc @cheftako

Reply:
> I see an error from a previous invocation of the http-connect test.
> 
> `Error from server: error dialing backend: parse "10.40.0.3:10250": first path segment in URL cannot contain colon`
> 
> [Link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/101059/pull-kubernetes-e2e-gce-network-proxy-http-connect/1382080868213854208/)
> 
> Is this related to the URL parsing?

After looking into the code in http/transport/roundtrip, the  format of addr passed to the dialor is always "host:port", so I think no need to parse it, just fix the Host addr directly based on the original line.

And the `pull-kubernetes-e2e-gce-network-proxy-http-connect` job has passed, PTAL. cc @Jefftree @cheftako

Reply:
Can the Host addr field be deleted, or just leave it here? Beacause it's same as request Host now.